ddtrace/internal/datadog/profiling/cmake/AnalysisFunc.cmake

```diff
@@ -5,36 +5,67 @@ function(add_ddup_config target)
+        # it until C++20.
+        target_compile_features(${target} PUBLIC cxx_std_20)
+        target_compile_options(
+            ${target}
+            PRIVATE "$<$<CONFIG:Release>:/O1>" # Optimize for size (equivalent to -Os)
+                    /Gy # Enable function-level linking (similar to -ffunction-sections)
+                    /W4 # Highest warning level (equivalent to -Wall -Wextra)
+                    /WX # Treat warnings as errors (equivalent to -Werror)
+                    /w44265 # Warn on shadowed variables (equivalent to -Wshadow)
+                    /w44287 # Warn on missing virtual destructors (equivalent to -Wnon-virtual-dtor)
+                    /w44412 # Warn on old-style casts (equivalent to -Wold-style-cast)
+                    /wd4201 # Don't treat the following as an error C4201: nonstandard extension used: nameless
+                            # struct/union
+                    /wd4267 # conversion from 'size_t' to 'unsigned short', possible loss of data
+                    /wd4996 # 'getpid': The POSIX name for this item is deprecated. Instead, use the ISO C and C++
+                            # conformant name: _getpid.
+                    /wd4100 # '_unusedop': unreferenced formal parameter
+                    /wd4245 # conversion from 'int' to 'uint64_t'
+                    /wd4244 # conversion from 'int64_t' to 'int'
+                    /wd4551)
     else()
-        # Non-macOS (e.g., Linux) options
-        target_compile_options(${target} PRIVATE "$<$<CONFIG:Debug>:-Og;-ggdb3>"
-                                                 "$<$<CONFIG:RelWithDebInfo>:-Os;-ggdb3>" -fno-semantic-interposition)
+        # Common compile options
+        target_compile_options(
+            ${target}
+            PRIVATE "$<$<CONFIG:Release>:-Os>"
+                    -ffunction-sections
+                    -Wall
+                    -Werror
+                    -Wextra
+                    -Wshadow
+                    -Wnon-virtual-dtor
+                    -Wold-style-cast)
+
+        if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+            # macOS-specific options
+            target_compile_options(${target} PRIVATE "$<$<CONFIG:Debug>:-Og;-g>" "$<$<CONFIG:RelWithDebInfo>:-Os;-g>")
+        else()
+            # Non-macOS (e.g., Linux) options
+            target_compile_options(
+                ${target} PRIVATE "$<$<CONFIG:Debug>:-Og;-ggdb3>" "$<$<CONFIG:RelWithDebInfo>:-Os;-ggdb3>"
+                                  -fno-semantic-interposition)
+        endif()
     endif()
 
     # Common link options
     target_link_options(${target} PRIVATE "$<$<CONFIG:RelWithDebInfo>:>")
 
-    if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+    if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
+        target_link_options(
+            ${target} PRIVATE "$<$<CONFIG:Release>:/OPT:REF>" # Equivalent to --gc-sections
+            "$<$<CONFIG:Release>:/INCREMENTAL:NO>" # Disable incremental linking (similar to -s for stripping symbols)
```

Comment:
In the comment, I'd clarify that this is used to decrease final size, but incremental linking isn't really equivalent to symbol-stripping AFAIK--it's related to whether updatable targets are created at each build stage.

incidentally, there are other options that might drive binary size, like /OPT:ICF (probably not very relevant for us) and possibly /DEBUG:NONE, but I believe that'll be implied